htlcswitch/payment_result.go

```diff
@@ -258,3 +258,48 @@ func fetchResult(tx kvdb.RTx, pid uint64) (*networkResult, error) {
 
 	return deserializeNetworkResult(r)
 }
+
+// cleaStore removes all entries from the store, except the payment IDs given.
+// NOTE: Since eery result not listed in the keep map will be deleted, care
+// should be taken to ensure no new payment attempts are being made
+// concurrently while this process is ongoing, as its result might end up being
+// deleted.
+func (store *networkResultStore) cleanStore(keep map[uint64]struct{}) error {
+	return kvdb.Update(store.db.Backend, func(tx kvdb.RwTx) error {
+		networkResults, err := tx.CreateTopLevelBucket(
+			networkResultStoreBucketKey,
+		)
+		if err != nil {
+			return err
+		}
+
+		var toClean [][]byte
+		if err := networkResults.ForEach(func(k, _ []byte) error {
```

Comment:
This could be further simplified by using a `kvdb.RwCursor`. Using the cursor there's no need to collect and delete but items can be deleted on the fly.

Reply:
Nice, didn't know about that! Done 👍